### PR TITLE
Rubocop/1.14.0: Upgrade rubocop to support ruby version 3.1

### PR DESCRIPTION
**Pull request by anandrikka**
## What

`snap-style` isn't compatible with Ruby 3.1:

```
Error: RuboCop found unknown Ruby version 3.1 in `.ruby-version`.
Supported versions: 2.3, 2.4, 2.5, 2.6, 2.7, 3.1
```

## Why
Version constraint defined on the `robocop` dependency is too strict and doesn't allow more recent versions.

Thanks to @orenmagid, to disable new rules temporarily the below list can help.
<details>
<summary>New Rules</summary>
<pre>
Layout/BeginEndAlignment: # (new in 0.91)
  Enabled: false
Layout/EmptyLinesAroundAttributeAccessor: # (new in 0.83)
  Enabled: false
Layout/SpaceAroundMethodCallOperator: # (new in 0.82)
  Enabled: false
Lint/BinaryOperatorWithIdenticalOperands: # (new in 0.89)
  Enabled: false
Lint/ConstantDefinitionInBlock: # (new in 0.91)
  Enabled: false
Lint/DeprecatedOpenSSLConstant: # (new in 0.84)
  Enabled: false
Lint/DuplicateElsifCondition: # (new in 0.88)
  Enabled: false
Lint/DuplicateRequire: # (new in 0.90)
  Enabled: false
Lint/DuplicateRescueException: # (new in 0.89)
  Enabled: false
Lint/EmptyConditionalBody: # (new in 0.89)
  Enabled: false
Lint/EmptyFile: # (new in 0.90)
  Enabled: false
Lint/FloatComparison: # (new in 0.89)
  Enabled: false
Lint/HashCompareByIdentity: # (new in 0.93)
  Enabled: false
Lint/IdentityComparison: # (new in 0.91)
  Enabled: false
Lint/MissingSuper: # (new in 0.89)
  Enabled: false
Lint/MixedRegexpCaptureTypes: # (new in 0.85)
  Enabled: false
Lint/OutOfRangeRegexpRef: # (new in 0.89)
  Enabled: false
Lint/RedundantSafeNavigation: # (new in 0.93)
  Enabled: false
Lint/SelfAssignment: # (new in 0.89)
  Enabled: false
Lint/TopLevelReturnWithArgument: # (new in 0.89)
  Enabled: false
Lint/TrailingCommaInAttributeDeclaration: # (new in 0.90)
  Enabled: false
Lint/UnreachableLoop: # (new in 0.89)
  Enabled: false
Lint/UselessMethodDefinition: # (new in 0.90)
  Enabled: false
Lint/UselessTimes: # (new in 0.91)
  Enabled: false
Style/AccessorGrouping: # (new in 0.87)
  Enabled: false
Style/BisectedAttrAccessor: # (new in 0.87)
  Enabled: false
Style/CaseLikeIf: # (new in 0.88)
  Enabled: false
Style/ClassEqualityComparison: # (new in 0.93)
  Enabled: false
Style/CombinableLoops: # (new in 0.90)
  Enabled: false
Style/ExplicitBlockArgument: # (new in 0.89)
  Enabled: false
Style/ExponentialNotation: # (new in 0.82)
  Enabled: false
Style/GlobalStdStream: # (new in 0.89)
  Enabled: false
Style/HashAsLastArrayItem: # (new in 0.88)
  Enabled: false
Style/HashLikeCase: # (new in 0.88)
  Enabled: false
Style/KeywordParametersOrder: # (new in 0.90)
  Enabled: false
Style/OptionalBooleanParameter: # (new in 0.89)
  Enabled: false
Style/RedundantAssignment: # (new in 0.87)
  Enabled: false
Style/RedundantFetchBlock: # (new in 0.86)
  Enabled: false
Style/RedundantFileExtensionInRequire: # (new in 0.88)
  Enabled: false
Style/RedundantRegexpCharacterClass: # (new in 0.85)
  Enabled: false
Style/RedundantRegexpEscape: # (new in 0.85)
  Enabled: false
Style/RedundantSelfAssignment: # (new in 0.90)
  Enabled: false
Style/SingleArgumentDig: # (new in 0.89)
  Enabled: false
Style/SlicingWithRange: # (new in 0.83)
  Enabled: false
Style/SoleNestedConditional: # (new in 0.89)
  Enabled: false
Style/StringConcatenation: # (new in 0.89)
  Enabled: false
</pre>
</details>

## How
Relax constraints to allow the use of a newer version supporting Ruby 3.1.

### CODE REVIEW

**Comment by anandrikka** [comment]
@PeterLanglois @santi-h @lluchez @nrgonzalez777

I appreciate some reviews on this PR. Dispatch is trying to upgrade the ruby version to `3.1` and `robocop` is a blocker.

**Comment by anandrikka** [comment]
I followed the previous PR #9

**Review by lluchez — APPROVED** [review]
Tested on Guidelines and it worked fine:

image

LG 👍

**Review by orenmagid — APPROVED** [review]
LGTM. We **could** explicitly disable the new cops that are pending by default to get rid of this warning:

```
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file.

Please also note that you can opt-in to new cops by default by adding this to your config:
  AllCops:
    NewCops: enable

Gemspec/DateAssignment: # new in 1.10
  Enabled: true
Layout/LineEndStringConcatenationIndentation: # new in 1.18
  Enabled: true
Layout/SpaceBeforeBrackets: # new in 1.7
  Enabled: true
Lint/AmbiguousAssignment: # new in 1.7
  Enabled: true
Lint/AmbiguousOperatorPrecedence: # new in 1.21
  Enabled: true
Lint/AmbiguousRange: # new in 1.19
  Enabled: true
Lint/DeprecatedConstants: # new in 1.8
  Enabled: true
Lint/DuplicateBranch: # new in 1.3
  Enabled: true
Lint/DuplicateRegexpCharacterClassElement: # new in 1.1
  Enabled: true
Lint/EmptyBlock: # new in 1.1
  Enabled: true
Lint/EmptyClass: # new in 1.3
  Enabled: true
Lint/EmptyInPattern: # new in 1.16
  Enabled: true
Lint/IncompatibleIoSelectWithFiberScheduler: # new in 1.21
  Enabled: true
Lint/LambdaWithoutLiteralBlock: # new in 1.8
  Enabled: true
Lint/NoReturnInBeginEndBlocks: # new in 1.2
  Enabled: true
Lint/NumberedParameterAssignment: # new in 1.9
  Enabled: true
Lint/OrAssignmentToConstant: # new in 1.9
  Enabled: true
Lint/RedundantDirGlobSort: # new in 1.8
  Enabled: true
Lint/RequireRelativeSelfPath: # new in 1.22
  Enabled: true
Lint/SymbolConversion: # new in 1.9
  Enabled: true
Lint/ToEnumArguments: # new in 1.1
  Enabled: true
Lint/TripleQuotes: # new in 1.9
  Enabled: true
Lint/UnexpectedBlockArity: # new in 1.5
  Enabled: true
Lint/UnmodifiedReduceAccumulator: # new in 1.1
  Enabled: true
Security/IoMethods: # new in 1.22
  Enabled: true
Style/ArgumentsForwarding: # new in 1.1
  Enabled: true
Style/CollectionCompact: # new in 1.2
  Enabled: true
Style/DocumentDynamicEvalDefinition: # new in 1.1
  Enabled: true
Style/EndlessMethod: # new in 1.8
  Enabled: true
Style/HashConversion: # new in 1.10
  Enabled: true
Style/HashExcept: # new in 1.7
  Enabled: true
Style/IfWithBooleanLiteralBranches: # new in 1.9
  Enabled: true
Style/InPatternThen: # new in 1.16
  Enabled: true
Style/MultilineInPatternThen: # new in 1.16
  Enabled: true
Style/NegatedIfElseCondition: # new in 1.2
  Enabled: true
Style/NilLambda: # new in 1.3
  Enabled: true
Style/NumberedParameters: # new in 1.22
  Enabled: true
Style/NumberedParametersLimit: # new in 1.22
  Enabled: true
Style/QuotedSymbols: # new in 1.16
  Enabled: true
Style/RedundantArgument: # new in 1.4
  Enabled: true
Style/RedundantSelfAssignmentBranch: # new in 1.19
  Enabled: true
Style/SelectByRegexp: # new in 1.22
  Enabled: true
Style/StringChars: # new in 1.12
  Enabled: true
Style/SwapValues: # new in 1.1
  Enabled: true
For more information: https://docs.rubocop.org/rubocop/versioning.html
```

We **could** also disable the cops that were enabled in Rubocop 1.0.0 if we're worried about new failures:

```
Layout/BeginEndAlignment: # (new in 0.91)
  Enabled: false
Layout/EmptyLinesAroundAttributeAccessor: # (new in 0.83)
  Enabled: false
Layout/SpaceAroundMethodCallOperator: # (new in 0.82)
  Enabled: false
Lint/BinaryOperatorWithIdenticalOperands: # (new in 0.89)
  Enabled: false
Lint/ConstantDefinitionInBlock: # (new in 0.91)
  Enabled: false
Lint/DeprecatedOpenSSLConstant: # (new in 0.84)
  Enabled: false
Lint/DuplicateElsifCondition: # (new in 0.88)
  Enabled: false
Lint/DuplicateRequire: # (new in 0.90)
  Enabled: false
Lint/DuplicateRescueException: # (new in 0.89)
  Enabled: false
Lint/EmptyConditionalBody: # (new in 0.89)
  Enabled: false
Lint/EmptyFile: # (new in 0.90)
  Enabled: false
Lint/FloatComparison: # (new in 0.89)
  Enabled: false
Lint/HashCompareByIdentity: # (new in 0.93)
  Enabled: false
Lint/IdentityComparison: # (new in 0.91)
  Enabled: false
Lint/MissingSuper: # (new in 0.89)
  Enabled: false
Lint/MixedRegexpCaptureTypes: # (new in 0.85)
  Enabled: false
Lint/OutOfRangeRegexpRef: # (new in 0.89)
  Enabled: false
Lint/RedundantSafeNavigation: # (new in 0.93)
  Enabled: false
Lint/SelfAssignment: # (new in 0.89)
  Enabled: false
Lint/TopLevelReturnWithArgument: # (new in 0.89)
  Enabled: false
Lint/TrailingCommaInAttributeDeclaration: # (new in 0.90)
  Enabled: false
Lint/UnreachableLoop: # (new in 0.89)
  Enabled: false
Lint/UselessMethodDefinition: # (new in 0.90)
  Enabled: false
Lint/UselessTimes: # (new in 0.91)
  Enabled: false
Style/AccessorGrouping: # (new in 0.87)
  Enabled: false
Style/BisectedAttrAccessor: # (new in 0.87)
  Enabled: false
Style/CaseLikeIf: # (new in 0.88)
  Enabled: false
Style/ClassEqualityComparison: # (new in 0.93)
  Enabled: false
Style/CombinableLoops: # (new in 0.90)
  Enabled: false
Style/ExplicitBlockArgument: # (new in 0.89)
  Enabled: false
Style/ExponentialNotation: # (new in 0.82)
  Enabled: false
Style/GlobalStdStream: # (new in 0.89)
  Enabled: false
Style/HashAsLastArrayItem: # (new in 0.88)
  Enabled: false
Style/HashLikeCase: # (new in 0.88)
  Enabled: false
Style/KeywordParametersOrder: # (new in 0.90)
  Enabled: false
Style/OptionalBooleanParameter: # (new in 0.89)
  Enabled: false
Style/RedundantAssignment: # (new in 0.87)
  Enabled: false
Style/RedundantFetchBlock: # (new in 0.86)
  Enabled: false
Style/RedundantFileExtensionInRequire: # (new in 0.88)
  Enabled: false
Style/RedundantRegexpCharacterClass: # (new in 0.85)
  Enabled: false
Style/RedundantRegexpEscape: # (new in 0.85)
  Enabled: false
Style/RedundantSelfAssignment: # (new in 0.90)
  Enabled: false
Style/SingleArgumentDig: # (new in 0.89)
  Enabled: false
Style/SlicingWithRange: # (new in 0.83)
  Enabled: false
Style/SoleNestedConditional: # (new in 0.89)
  Enabled: false
Style/StringConcatenation: # (new in 0.89)
  Enabled: false
```
I'm fine with it either way.

**Comment by anandrikka** [comment]
@orenmagid I disabled most of them to keep changes to a minimum in our upgrade [PR](https://github.com/bodyshopbidsdotcom/dispatch/blob/e5d5b596ee1868f84a9e55bc7aa1ea74714f9d17/.rubocop.yml#L44), but eventually want to add them back, blocking them at the gem level might increase work for application?

**Comment by orenmagid** [comment]
> @orenmagid I disabled most of them to keep changes to a minimum in our upgrade [PR](https://github.com/bodyshopbidsdotcom/dispatch/blob/e5d5b596ee1868f84a9e55bc7aa1ea74714f9d17/.rubocop.yml#L44), but eventually want to add them back, blocking them at the gem level might increase work for application?

Yeah, fair enough. I could see disabling at either level. Perhaps folks from the backend committee could give their opinion.

**Comment by PeterLanglois** [comment]
There's debate as to whether or not we'll keep this gem, so we'd have to circle back on this.

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,4 @@
 
 # rspec failure tracking
 .rspec_status
+.idea
```
Comment (PeterLanglois) [inline .gitignore]: appease GH and add the new line!

**Review by PeterLanglois — APPROVED** [review]

**Pull request merged**